manifests/rule.pp

```diff
@@ -26,6 +26,22 @@
 # @param content
 #   The **exact** content of the rule that should be added
 #
+#   * If this module is run in ``firewalld`` mode, this needs to contain a JSON
```

Comment:
This looks like a breaking change to me.  I think a new defined type should have been added instead iptables::firewalld_rule

Reply:
That seems reasonable.

Reply:
@lnemsick-simp Your suggestion has turned out to be 100% better than the original implementation in all ways. Thanks!